datajunction-server/datajunction_server/construction/build_v2.py

```diff
@@ -179,202 +184,398 @@ async def get_measures_query(  # pylint: disable=too-many-locals
-                dimension_join.node_query = cte_mapping[link.dimension.name]
-                continue
+    async def build(self) -> ast.Query:
+        """
+        Builds the node SQL with the requested set of dimensions, filter expressions,
+        order by, and limit clauses.
+
+        Build Strategy
+        ---------------
+        1. Recursively turn node references into query ASTs + apply any filters that can
+        be pushed down. If the node query has CTEs, unwind them into subqueries.
+        2. Initialize the final query with the node's query AST added to it as a CTE.
+        3. For any dimensions or filters requested for the node, determine if a join is
+        needed to bring in the dimension or filter. Keep track of all the necessary dimension
+        joins in a dict that maps dimension nodes to join paths.
+        4. For each of the necessary dimension joins, build the dimension node's query in the
+        same manner as above, recursively replacing any node references and pushing down requested
+        filters where possible.
+        5. Add each dimension node's query AST to the final query as a CTE.
+        6. Build the final query using the various CTEs. This does all the joins between the node
+        query AST and the dimension nodes' ASTs using the join logic from the dimension links.
+        7. Add all requested dimensions to the final select.
+        8. Add order by and limit to the final select (TODO)
+        """
+        physical_table = get_table_for_node(
+            self.node_revision,
+            build_criteria=self._build_criteria,
+        )
+        if physical_table and not self._filters and not self.dimensions:
+            physical_table.parenthesized = False
+            return ast.Query(
+                select=ast.Select(
+                    projection=physical_table.columns,  # type: ignore
+                    from_=ast.From(relations=[ast.Relation(physical_table)]),
+                ),
+            )
 
-            dimension_node_query = await build_dimension_node_query(
-                session,
-                build_criteria,
-                link,
-                to_filter_asts(filters),
-                cte_mapping,
+        node_ast = (
+            await compile_node_ast(self.session, self.node_revision)
+            if not physical_table
+            else ast.Query(
+                select=ast.Select(
+                    projection=physical_table.columns,  # type: ignore
+                    from_=ast.From(relations=[ast.Relation(physical_table)]),
+                ),
             )
-            dimension_join.node_query = convert_to_cte(
-                dimension_node_query,
-                final_ast,
-                link.dimension.name,
+        )
+
+        # Find all dimension node joins necessary for the requested dimensions and filters
+        dimension_node_joins = await self.find_dimension_node_joins()
+
+        ###################
+        ## Build Node CTEs
```

Comment:
I like that this comment highlights this step, but why not make it self documenting by converting this to a build step method and just calling `self.build_node_ctes()` here?